src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -2,77 +2,60 @@
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.StackTraceExplorer
 {
     /// <summary>
-    /// A line of text that was parsed by <see cref="StackTraceAnalyzer" />
-    /// to provide metadata bout the line. Expected to be the parsed output 
-    /// of a serialized <see cref="StackFrame"/>
+    /// A line from <see cref="StackTraceAnalyzer.Parse(string, CancellationToken)"/> that
+    /// was parsed by <see cref="StackFrameParser"/>
     /// </summary>
     internal sealed class ParsedStackFrame : ParsedFrame
     {
+        public readonly StackFrameTree Tree;
+
         public ParsedStackFrame(
-            string originalText,
-            TextSpan typeSpan,
-            TextSpan methodSpan,
-            TextSpan argsSpan,
-            TextSpan fileSpan = default)
-            : base(originalText)
+            StackFrameTree tree)
         {
-            Contract.ThrowIfTrue(typeSpan.IsEmpty);
-            Contract.ThrowIfTrue(methodSpan.IsEmpty);
-
-            TypeSpan = typeSpan;
-            MethodSpan = methodSpan;
-            ArgsSpan = argsSpan;
-            FileSpan = fileSpan;
+            Tree = tree;
         }
 
-        /// <summary>
-        /// The full type name parsed from the line. 
-        /// ex: [|Microsoft.CodeAnalysis.Editor.CallstackExplorer.|]Example(arg1, arg2)
-        /// </summary>
-        public TextSpan TypeSpan { get; }
-
-        /// <summary>
-        /// The method name span
-        /// ex: Microsoft.CodeAnalysis.Editor.CallstackExplorer.[|Example|](arg1, arg2)
-        /// </summary>
-        public TextSpan MethodSpan { get; }
+        public StackFrameCompilationUnit Root => Tree.Root;
 
-        /// <summary>
-        /// The span of comma seperated arguments.
-        /// ex: Microsoft.CodeAnalysis.Editor.CallstackExplorer.Example[|(arg1, arg2)|]
-        /// </summary>
-        public TextSpan ArgsSpan { get; }
-
-        /// <summary>
-        /// The span representing file information on the stack trace line. Is not always available, so it's 
-        /// possible this span is <see langword="default"/>
-        /// </summary>
-        public TextSpan FileSpan { get; }
+        public override string ToString()
+        {
+            return Tree.Text.CreateString();
+        }
 
         public async Task<ISymbol?> ResolveSymbolAsync(Solution solution, CancellationToken cancellationToken)
         {
-            // The original span for type includes the trailing '.', which we don't want when
-            // looking for the class by metadata name
-            var fullyQualifiedTypeName = OriginalText[TypeSpan.Start..(TypeSpan.End - 1)];
-            var methodName = GetMethodText();
+            // MemberAccessExpression is [Expression].[Identifier], and Identifier is the 
+            // method name.
+            var typeExpression = Root.MethodDeclaration.MemberAccessExpression.Left;
+            var fullyQualifiedTypeName = typeExpression.ToString(skipTrivia: true);
```

Comment:
i don't love .ToString of this form being on the node itself.  would prefer a helper here if this is the only place needed.